sys/filedesc.c

```diff
@@ -87,9 +120,11 @@ fdtab_t *fdtab_copy(fdtab_t *fdt) {
 
   mtx_scoped_lock(&fdt->fdt_mtx);
 
-  /* We can assume both filedescs use 20 descriptors,
-   * because we don't support other numbers. */
-  assert(fdt->fdt_nfiles == newfdt->fdt_nfiles);
+  if (fdt->fdt_nfiles > newfdt->fdt_nfiles) {
+    mtx_lock(&newfdt->fdt_mtx);
```

Comment:
A side note: No other threads have a reference to `newfdt` yet, so locking the mutex should be unnecessary. However, `fd_growtable` expects to have the mutex already locked, and therefore we need to lock it before growing the table. I see some redundancy here, but there is no easy way to solve this - so I suggest to leave it as it is (or maybe just a comment on why is `mtx_lock` necessary here).